frontend/src/stores/user-store.ts

```diff
@@ -25,6 +35,50 @@ const initialUserObject = {
 export const useUserStore = defineStore('user', () => {
   const data = useLocalStorage('tba/user', structuredClone(initialUserObject));
 
+  const call = ref(null);
+
+  /**
+   * Initialize store with data required at runtime
+   * 
+   * @param fetch preconfigured function to perform API calls
+   */
+  const init = (fetch: Fetch) => {
+    call.value = fetch;
+  }
+
+  // Init user config if not already available
+  if (!data.value?.settings) {
+    const dj = inject(dayjsKey);
+    const detectedTimeFormat = Number(dj('2022-05-24 20:00:00').format('LT').split(':')[0]) > 12 ? 24 : 12;    
+
+    data.value.settings = {
+      language: i18n.locale.value,
+      colorScheme: ColorSchemes.System,
+      timeFormat: detectedTimeFormat,
+      timezone: dj.tz.guess(),
+    };
+  }
```

Comment:
This handles the new settings location for existing users already having a local copy of the user store